### PR TITLE
Cross-Compiler Documentation Update + Minor CodeQL Alert Resolutions

### PR DESCRIPTION
**Pull request by Boehm-Michael**
| | |
|:---|:---|
|**_Originating Project/Creator_**| boehm-Michael |
|**_Affected Component_**| N/A |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**| CodeQL |
|**_Has Unit Tests (y/n)_**| N/A |
|**_Builds Without Errors (y/n)_**| Y |
|**_Unit Tests Pass (y/n)_**| N/A |
|**_Documentation Included (y/n)_**| Y |

---
## Change Description

Updated cross-compiler documentation and fixed code readability as noted by CodeQL

## Rationale

Resolve CodeQL alerts, fix broken cross-compiler documentation

## Testing/Review Recommendations

N/A

## Future Work

Continue to update for CodeQL alerts.

Fixed PR of #2096

### CODE REVIEW

File: Svc/TlmChan/test/ut/Tester.hpp [hunk 1]
```diff
@@ -42,21 +42,15 @@ class Tester : public TlmChanGTestBase {
...
     //! Handler for from_PktSend
     //!
-    void from_PktSend_handler(const NATIVE_INT_TYPE portNum, /*!< The port number*/
-                              Fw::ComBuffer& data,           /*!<
-                                    Buffer containing packet data
-                                    */
-                              U32 context                    /*!<
-                                             Call context value; meaning chosen by user
-                                             */
+    void from_PktSend_handler(const NATIVE_INT_TYPE portNum, //!< The port number*/
```
Comment (LeStarch) [inline Svc/TlmChan/test/ut/Tester.hpp:45]: Missed the closing `*/`

File: Ref/TypeDemo/TypeDemo.hpp [hunk 2]
```diff
@@ -38,78 +38,70 @@ class TypeDemo : public TypeDemoComponentBase {
...
-                                                   U8 repeat,                 /*!<
-                                                                                   Number of times to repeat the choices
-                                                                                   */
-                                                   Ref::ChoiceSlurry choices, /*!<
-                                                   A phenomenal amount of choice
-                                                   */
-                                                   U8 repeat_max              /*!<
-                                                                             Limit to the number of repetitions
-                                                                             */
+    void GLUTTON_OF_CHOICE_WITH_FRIENDS_cmdHandler(const FwOpcodeType opCode, //!< The opcode*/
```
Comment (LeStarch) [inline Ref/TypeDemo/TypeDemo.hpp:94]: Missed closing `*/`

File: Ref/TypeDemo/TypeDemo.hpp [hunk 3]
```diff
@@ -38,78 +38,70 @@ class TypeDemo : public TypeDemoComponentBase {
...
+                                                   const U32 cmdSeq,          //!< The command sequence number*/
```
Comment (LeStarch) [inline Ref/TypeDemo/TypeDemo.hpp:95]: Missed closing `*/`

**Review by LeStarch — CHANGES_REQUESTED** [review]
There are a few `*/` ending comments that remain we should clean those up.

Also, I was hoping we could pull all these instructions into a separate package.  e.g. `docs/Tutorials/CrossCompilationSetup` where we talk through installing each peice and the toolchain as a separate pseudo-tutorial that would be linked from exerywhere.

Do you want to refactor that or should I?

File: Ref/TypeDemo/TypeDemo.hpp (hunk 3 above)
Comment (jwest115) [reply Ref/TypeDemo/TypeDemo.hpp]: Fixed!

File: Ref/TypeDemo/TypeDemo.hpp (hunk 2 above)
Comment (jwest115) [reply Ref/TypeDemo/TypeDemo.hpp]: Fixed!

File: Svc/TlmChan/test/ut/Tester.hpp [hunk 4]
```diff
@@ -42,21 +42,15 @@ class Tester : public TlmChanGTestBase {
...
+                              U32 context                    //!< Call context value; meaning chosen by user
     );
 
     //! Handler for from_pingOut
     //!
-    void from_pingOut_handler(const NATIVE_INT_TYPE portNum, /*!< The port number*/
-                              U32 key                        /*!<
-                                                 Value to return to pinger
-                                                 */
+    void from_pingOut_handler(const NATIVE_INT_TYPE portNum, //!< The port number*/
```
[the remark this answers is not in the document]
Comment (jwest115) [reply Svc/TlmChan/test/ut/Tester.hpp]: Fixed!

**Comment by jwest115** [comment]
@LeStarch I spit the documentation into different sections:
- Cross Compilation Setup
- Compiling for ARM
- Running on ARM Linux

All of the above have been added to the "CrossCompilationSetup" directory, which can be referenced in other tutorials.

Note: I did remove the original "Cross-Compilation Tutorial", since it has been split into multiple sections. I can add it back if you think we need it. 

As for next steps, we may want to go through the other tutorial to make sure things are linked properly.

File: docs/Tutorials/CrossCompilationSetup/ArmLinuxTutorial.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+```sh
+# For in-person workshops and ARM 64-bit hardware
+# In: Deployment Folder
+fprime-gds -n --dictionary build-artifacts/aarch64-linux/dict/<App Dictionary>.xml
+
+# For ARM 32-bit hardware
+# In: Deployment Folder
+fprime-gds -n --dictionary build-artifacts/aarch64-linux/dict/<App Dictionary>.xml
+```
+
```
Comment (LeStarch) [inline docs/Tutorials/CrossCompilationSetup/ArmLinuxTutorial.md:30]: We should add a step noting one should also SSH into the hardware to run the binary.

File: docs/Tutorials/CrossCompilationSetup/CrossCompilationSetupTutorial.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+3. 5GB of free disk space, 8 GB of RAM
+4. Knowledge of the command line for your operating system (Bash, Powershell, Zsh, etc).
+
+<a name="Installing Dependencies"></a>
+## 2. Installing Dependencies
+
+Choose the operating system you are using to install F Prime:
+
+- [Windows 10/11 WSL](./Windows.md)
+- [Mac OS X](./Mac.md)
```
Comment (LeStarch) [inline docs/Tutorials/CrossCompilationSetup/CrossCompilationSetupTutorial.md:32]: You link to Mac.md here, but I don't see that in our list of files.

**Review by LeStarch — CHANGES_REQUESTED** [review]

File: docs/Tutorials/CrossCompilationSetup/ArmLinuxTutorial.md (hunk 5 above)
Comment (jwest115) [reply docs/Tutorials/CrossCompilationSetup/ArmLinuxTutorial.md:30]: Added step for SSHing into device and running the executable

File: docs/Tutorials/CrossCompilationSetup/CrossCompilationSetupTutorial.md (hunk 6 above)
Comment (jwest115) [reply docs/Tutorials/CrossCompilationSetup/CrossCompilationSetupTutorial.md]: Added file

**Review by LeStarch — APPROVED** [review]

**Review by LeStarch — APPROVED** [review]

**Pull request merged**